packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -191,14 +188,21 @@ export class CdkToolkit {
           }
         }
 
+        const diff = options.securityOnly
+          ? fullDiff(currentTemplate, stack.template, changeSet)
+          : fullDiff(currentTemplate, stack.template, changeSet, !!resourcesToImport);
```

Comment:
We should not be calling `fullDiff` across both this method and `printSecurityDiff`. I'd prefer we keep `fullDiff` in `printSecurityDiff`, and then move the stack print to `printSecurityDiff`.